pkg/provisioningv2/rke2/planner/planner.go

```diff
@@ -862,7 +862,8 @@ func getTaints(entry *planEntry) (result []corev1.Taint, _ error) {
 	}
 
 	if !isWorker(entry) {
-		if isEtcd(entry) {
+		// k3s charts do not have correct tolerations when the master node is both controlplane and etcd
+		if isEtcd(entry) && (rke2.GetRuntime(cp.Spec.KubernetesVersion) != rke2.RuntimeK3S || !isControlPlane(entry)) {
```

Comment:
If all we need from CP here is `cp.Spec.KubernetesVersion` why not pass just it (or even `isK3s` flag - but I like former more anyway) and not the whole object (even though it's a pointer).

Reply:
Right, actually thinking more about it passing `string runtime` looks the cleanest to me and matching what we've had before.
I'm not so sure about the "remove in the next release" if the fix has to be on k3s side considering we will be supporting previous k3s patch versions still.
Probably splitting hairs here, so I'm not going to block this based on whether we should pass the whole CP or just string runtime.